### ISSUE TITLE
utilities.m4:1: warning: file `list.m4' included several times ...

### ISSUE THREAD

**Issue by bs27975** [matejak/argbash#91]
(Debian stretch.)
Having followed [User installation](https://argbash.readthedocs.io/en/stable/install.html)

The top bit of 'make check' returns:
```/usr/local/src/argbash-2.8.0/resources$ make check
make unittests
make[1]: Entering directory '/usr/local/src/argbash-2.8.0/resources'
autom4te -l m4sugar -I ../src/ -I ../tests/unittests ../tests/unittests/check-utils.m4 && autom4te -l m4sugar -I ../src/ -I ../tests/unittests ../tests/unittests/check-indentation.m4 && autom4te -l m4sugar -I ../src/ -I ../tests/unittests ../tests/unittests/check-list.m4 && autom4te -l m4sugar -I ../src/ -I ../tests/unittests ../tests/unittests/check-types.m4 && autom4te -l m4sugar -I ../src/ -I ../tests/unittests ../tests/unittests/check-arguments.m4 && true
../src/utilities.m4:1: warning: file `list.m4' included several times
../src/utilities.m4:1: warning: file `list.m4' included several times
../src/utilities.m4:1: warning: file `list.m4' included several times
../src/value_validators.m4:1: warning: file `function_generators.m4' included several times
make[1]: Leaving directory '/usr/local/src/argbash-2.8.0/resources'
```
Further down:
```test -z "shellcheck -x -e 2015" || shellcheck -x -e 2015 "../tests/regressiontests/call-salone.sh"

In ../tests/regressiontests/call-salone.sh line 15:
. "$script_dir/test-salone.sh"  # '.' means 'source'
^-- SC1090: Can't follow non-constant source. Use a directive to specify location.


In ../tests/regressiontests/call-salone.sh line 22:
echo "BOOL=$_arg_boo_l,OPT_S=$_arg_opt_arg,POS_S=$_arg_pos_arg,POS_OPT=$_arg_pos_opt,OPT_INCR=$_arg_opt_incr,"
           ^-- SC2154: _arg_boo_l is referenced but not assigned.
                             ^-- SC2154: _arg_opt_arg is referenced but not assigned.
                                                 ^-- SC2154: _arg_pos_arg is referenced but not assigned.
                                                                       ^-- SC2154: _arg_pos_opt is referenced but not assigned.
                                                                                              ^-- SC2154: _arg_opt_incr is referenced but not assigned.

../tests/regressiontests/Makefile:268: recipe for target 'call-salone' failed
make[1]: *** [call-salone] Error 1
make[1]: Leaving directory '/usr/local/src/argbash-2.8.0/resources'
Makefile:120: recipe for target 'check' failed
make: *** [check] Error 2
```

How do I address these?

<hr>

Using the sample of: 
```
--pos positional-arg --opt option --opt-bool print
```
from [Minimal example](https://argbash.readthedocs.io/en/latest/example.html#minimal-example) ultimately I get:
```
argbash \
        --check-typos --type bash-script \
        -c --strip all \
parsecmdline.m4 \
--output parsecmdline.bash
//usr/lib/argbash/utilities.m4:1: warning: file `list.m4' included several times
//usr/lib/argbash/value_validators.m4:1: warning: file `function_generators.m4' included several times
Error during autom4te run, aborting!
```
and no output.

What have I failed to set up correctly?

TIA!

**Comment by bs27975** [comment]
^-- SC1090: Can't follow non-constant source. Use a directive to specify location.
 ^-- SC2154: _arg_boo_l is referenced but not assigned., etc.

First seems to be solved by:
```
# shellcheck source=/usr/local/src/argbash-2.8.0/tests/regressiontests/test-salone.sh
```
above it.
\- never mind line 3 already has the same`

And the latters by:
```
# shellcheck disable=SC2154
```
And using `-x` on the call to shellcheck.

CORRECTION: make is already invoking it that way.
(But still useful for checking when in that directory, via 
`/usr/local/src/argbash-2.8.0/tests/regressiontests$ shellcheck -x call-salone.sh`
 )

This all matters as `make check` stops at the first failed check. Continuing after the above ...
```
In ../tests/regressiontests/test-onlypos.sh line 87:
echo "POS_S=$_arg_pos_arg,POS_OPT=$_arg_pos_opt,"
            ^-- SC2154: _arg_pos_arg is referenced but not assigned.
```
Added `# shellcheck disable=SC2154`

I am not saying these are necessary the correct answers for remediation back into source. I have not sunk deeply enough into things to be certain of that.

The shellcheck source line feels more like an invocation issue - duplicated line 3 on the line immediately above the error line solves it, when the invocation on line 3 is supposed to apply to the entire file. Perhaps line 2 may not be blank, for shellcheck's purposes.

Continuing, I experience:
```
In ../tests/regressiontests/test-standalone.sh line 39:
_arg_pos_opt="pos-default"
^-- SC2034: _arg_pos_opt appears unused. Verify it or export it.


In ../tests/regressiontests/test-standalone.sh line 72:
                                _arg_opt_arg="${_key##-o}"
                                ^-- SC2034: _arg_opt_arg appears unused. Verify it or export it.


In ../tests/regressiontests/test-standalone.sh line 84:
                                test "${1:0:5}" = "--no-" && _arg_boo_l="off"
                                                             ^-- SC2034: _arg_boo_l appears unused. Verify it or export it.

../tests/regressiontests/Makefile:337: recipe for target 'stability-salone' failed
```

Arguably, these errors are permissible (except why document 'make check' then - and thank you for doing so). I wouldn't know.

Except, argbash is failing for me, and my first supposition is that I have installed it incorrectly, or there is is something non-standard in how I have crafted my environment. So things like 'make check' are one's goto for sanity checking same.

I have since concluded that any argbash failure for me in 'argbash is failing for me' is user error. If it turns out to be more than that, I can report so back here in a different issue.

Cheers.

**Comment by bs27975** [comment]
Putting '-' before the make test lines, e.g.
```
 -test -z "$(SHELLCHECK)" || $(SHELLCHECK) $(ARGBASH_BIN)
```
is handy to at least see all the issues detected, in one go, without failing upon each one.

And when the output of 'make check' gets to be too much, 
```
make check | tee | grep "ERROR" > /tmp/makerror.txt ; less /tmp/makerror.txt
```
to reduce some background noise and better focus.

However, I do still get:
```
ERROR="namely: 'pos-arg' and 'pos-more'" ../tests/regressiontests/reverse ../tests/regressiontests/test-more.sh
ERROR=spurious ../tests/regressiontests/reverse ../tests/regressiontests/test-onlypos.sh one two three
ERROR='between 1 and 2' ../tests/regressiontests/reverse ../tests/regressiontests/test-onlypos.sh one two three
ERROR='Not enough' ../tests/regressiontests/reverse ../tests/regressiontests/test-onlypos.sh
ERROR=spurious ../tests/regressiontests/reverse ../tests/regressiontests/test-onlypos-declared.sh one two three
ERROR='between 1 and 2' ../tests/regressiontests/reverse ../tests/regressiontests/test-onlypos-declared.sh one two three
ERROR='Not enough' ../tests/regressiontests/reverse ../tests/regressiontests/test-onlypos-declared.sh
~
```

**Comment by matejak** [comment]
This is partially addressed in #94.

**Comment by matejak** [comment]
Thank you for reporting the issue, I have to say that I am not sure what the actual errors are. `make check` is supposed to pass without errors (it is also assured by CI), so stopping `make` at firs problem seems like a good decision to me.
I have fixed the duplicated include problems, so could you please try again?

**Comment by matejak** [comment]
I close this as I believe the issue has been fixed by #94.
